### PR TITLE
4227 update dev/test env usecase with demo notebooks and walkthrough video

### LINKED ISSUE #4227: Update usecases with demo notebooks and walkthrough video
1. Update the use cases docs with demo notebooks listed in https://github.com/treeverse/lakeFS-samples and link to walkthrough videos as well.

### PR DESCRIPTION
**Pull request by vinodhini8694**
Fix #4227

Updated the docs and added demo notebook reference as well.
TODO: Walkthrough video to be embedded.

### CODE REVIEW

File: docs/use_cases/iso_env.md [hunk 1]
```diff
@@ -1,116 +1,190 @@
...
+Running ETL and transformation jobs directly in production is a guaranteed way to have data issues flow into dashboards, ML models, and other consumers sooner or later. The most common approach to avoid making changes directly in production is to create and maintain multiple data environments. Dev environment to develop the data pipelines and test environment where pipeline changes are implemented before pushing it to production.
 
-The issue with this approach is that it's time-consuming and costly to maintain this separate test environment. And for larger teams it forces multiple people to share a single environment, requiring significant co-ordination.
+The issue with this approach is that it's time-consuming and costly to maintain these separate dev/test environments. And for larger teams it forces multiple people to share these environments, requiring significant co-ordination.
 
-## How do I create isolated environments with lakeFS?
+## How do I create dev/test environments with lakeFS?
```
Comment (adipolak) [inline docs/use_cases/iso_env.md:20]: "## How do I create an isolated data dev/test environment with lakeFS?

File: docs/use_cases/iso_env.md [hunk 2]
```diff
@@ -1,116 +1,190 @@
...
 
-To install the AWS CLI, follow [these instructions](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)
-{: .note .note-info }
+Follow along the steps below to create dev/test environment with lakeFS.
 
-Download the file
+* Start by cloning the repository:
 ```bash
-aws s3 cp s3://amazon-reviews-pds/parquet/product_category=Sports/part-00000-495c48e6-96d6-4650-aa65-3c36a3516ddd.c000.snappy.parquet $HOME/
+git clone https://github.com/treeverse/lakeFS-samples.git && cd 03-apache-spark-python-demo
```
Comment (adipolak) [inline docs/use_cases/iso_env.md:86]: love the usage of the sample repository!!!

**Review by adipolak — APPROVED** [review]
looks great! Added one suggestion. Thanks for this PR!

File: docs/use_cases/iso_env.md [hunk 3]
```diff
@@ -1,116 +1,190 @@
...
 
 The key difference when using lakeFS for isolated data environments is that you can create them immediately before testing a change. And once new data is merged into production, you can delete the branch - effectively deleting the old environment.
 
 This is different from creating a long-living test environment used as a staging area to test all the updates. With lakeFS, **we create a new branch for each change to production** that we want to make. One benefit of this is the ability to test multiple changes at one time.
 
+![dev/test branches as environments]({{ site.baseurl }}/assets/img/iso_env_dev_test_branching.png)
+
+# Creating Dev/Test Environments with lakeFS
+
+lakeFS supports UI, CLI (`lakectl` commandline utility) and several client API [integrations](../integrations) to run the git-like operations. Let us explore how to create dev/test environments using each of these options below.
```
Comment (johnnyaug) [inline docs/use_cases/iso_env.md:42]:
```suggestion
lakeFS supports UI, CLI (`lakectl` commandline utility) and several client API [integrations](../integrations) to run Git-like operations. Let us explore how to create dev/test environments using each of these options below.
```

File: docs/use_cases/iso_env.md [hunk 4]
```diff
@@ -1,116 +1,190 @@
...
+![sample repository]({{ site.baseurl }}/assets/img/iso_env_myrepo.png)
+
+
+Click on `my-repo` and notice that by default, the repo has a `main` branch created and `sample_data` preloaded to work with.
+
+
+![main branch]({{ site.baseurl }}/assets/img/iso_env_sampledata.png)
+
+
+You can create a new branch say `test-env` by going to the branches tab and clicking `Create branch`. Once it is successful, we see two branches under the repo: `main` and `test-env`.
```
Comment (johnnyaug) [inline docs/use_cases/iso_env.md:59]:
```suggestion
You can create a new branch (say, `test-env`) by going to the _Branches_ tab and clicking _Create Branch_. Once it is successful, you will see two branches under the repo: `main` and `test-env`.
```

File: docs/use_cases/iso_env.md [hunk 5]
```diff
@@ -1,116 +1,190 @@
...
+
+
+![test-env branch]({{ site.baseurl }}/assets/img/iso_env_testenv_branch.png)
+
+
+Now, you can add, modify or delete the data files in the `test-env` branch without affecting the data in the main branch.
```
Comment (johnnyaug) [inline docs/use_cases/iso_env.md:65]:
```suggestion
Now you can add, modify or delete objects under the `test-env` branch without affecting the data in the main branch.
```

File: docs/use_cases/iso_env.md [hunk 6]
```diff
@@ -1,116 +1,190 @@
...
+
+## Using lakeFS Python Client API and Jupyter notebook
+
+Adding or deleting files from different branches in a lakeFS repo can be performed from the UI as shown above. However, to perform complex transformations on the data using Spark or other compute engine, you would need to use lakeFS supported APIs.
```
Comment (johnnyaug) [inline docs/use_cases/iso_env.md:69]:
```suggestion
Adding or deleting objects to different branches in a lakeFS repository can be performed from the UI as shown above. However, to perform complex transformations using Spark or other compute engines, you will need to use lakeFS clients.
```

File: docs/use_cases/iso_env.md [hunk 7]
```diff
@@ -1,116 +1,190 @@
...
 
-Go to your Spark notebook UI, in our case this is the Jupyter UI.
+lakeFS can be configured to work with Spark in two ways:
+* Access lakeFS using the [S3A gateway](../integrations/spark.md#access-lakefs-using-the-s3a-gateway)
+* Access lakeFS using the [lakeFS-specific Hadoop FileSystem](../integrations/spark.md#access-lakefs-using-the-lakefs-specific-hadoop-filesystem)
 
-Inside the notebook, create a new notebook and start a spark context:
+### Upload the sample data to main branch
+
+To upload a file to the `my-repo`, use the following command.
```
Comment (johnnyaug) [inline docs/use_cases/iso_env.md:137]:
```suggestion
To upload an object to the `my-repo`, use the following command.
```

File: docs/use_cases/iso_env.md [hunk 8]
```diff
@@ -1,116 +1,190 @@
...
 ```
 
-Now we can use Spark to read in the Parquet file we added to the main branch of our lakeFS repo:
+Once uploaded, commit the changes to the `main` branch and attach some metadata to the commit as well.
+```bash
+client.commits.commit(
+    repository="my-repo",
+    branch="main",
+    commit_creation=models.CommitCreation(
+        message='Added my first file!',
```
Comment (johnnyaug) [inline docs/use_cases/iso_env.md:154]:
```suggestion
        message='Added my first object!',
```

File: docs/use_cases/iso_env.md [hunk 9]
```diff
@@ -1,116 +1,190 @@
...
+![test-env branch]({{ site.baseurl }}/assets/img/iso_env_testenv_branch.png)
+
+
+Now, you can add, modify or delete the data files in the `test-env` branch without affecting the data in the main branch.
+
+## Using lakeFS Python Client API and Jupyter notebook
+
+Adding or deleting files from different branches in a lakeFS repo can be performed from the UI as shown above. However, to perform complex transformations on the data using Spark or other compute engine, you would need to use lakeFS supported APIs.
+
+This use case shows how you can create dev/test data environments using lakeFS branches. The following tutorial will use an existing lakeFS environment (i.e., playground) and a jupyter notebook, python lakefs_client API to demonstrate integration of lakeFS with [Spark](../integrations/spark). You can run this tutorial on your local machine.
```
Comment (johnnyaug) [inline docs/use_cases/iso_env.md:71]: Not clear.

**Review by johnnyaug — CHANGES_REQUESTED** [review]

File: docs/use_cases/iso_env.md (hunk 9 above)
Comment (vinodhini8694) [reply docs/use_cases/iso_env.md]: Fixed it.

**Comment by vinodhini8694** [comment]
> I am not sure about repeating in our documentation the documentation of the samples-repo - but I am not the expert on this. I would ask for a second opinion though.


@iddoavn  We had a different example and code snippet in the docs and in the sample notebook. So I updated the docs to use the same snippet as lakeFS samples notebook.

**Review by adipolak — APPROVED** [review]

**Comment by vinodhini8694** [comment]
@johnnyaug @iddoavn Added the requested changes - except embedding the webinar video recording. Can we move forward?

**Comment by iddoavn** [comment]
@vinodhini8694 - why not embed the video?

**Review by iddoavn — CHANGES_REQUESTED** [review]
I think we should embed the video as it shows the end-to-end process of what is described here. 
We should start it on the point of the demo using this link: https://youtu.be/VtGNw2YPFEU?t=533

**Comment by vinodhini8694** [comment]
@iddoavn Embedded the video to the page as well.

**Pull request merged**